### ISSUE TITLE
Reproducibily of error -  default activity not found in AndroidManifest.xml

### ISSUE THREAD

**Issue by somghosh** [excilys/androidannotations#1290]
I am using android annotations in Android Studio, and I keep getting the error when running (after successful build), that Default Activity is not found. It comes and goes, and sometimes it just needs a restart of Android Studio, sometimes a clean and rebuild, and sometime it just does not go away (I have no idea of how to reproduce it consistently).

I understand that things are a bit buggy, but how to reproduce this error consistently, and what is a sure step to remove it each time it appears

        <activity
            android:name="com.example.somghosh.earthmiles.views.MainActivity_"
            android:label="@string/app_name" >
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />

                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>

        </activity>

###build.girdle

    buildscript {
        repositories {
            mavenCentral()

        }
        dependencies {
            classpath 'com.neenbedankt.gradle.plugins:android-apt:1.2+'

        }
    }

    apply plugin: 'com.android.application'
    apply plugin: 'android-apt'

    repositories {
        mavenCentral()
        maven {
            url "https://oss.sonatype.org/content/repositories/snapshots/"
        }
    }

    android {
        compileSdkVersion 21
        buildToolsVersion "21.1.1"

        defaultConfig {
            applicationId "com.example.somghosh.earthmiles"
            minSdkVersion 15
            targetSdkVersion 21
            versionCode 1
            versionName "1.0"
        }
        buildTypes {
            release {
                runProguard false
                proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
            }
        }
        packagingOptions {
            exclude 'META-INF/services/javax.annotation.processing.Processor'
        }
    }

    apt {
        arguments {
            androidManifestFile variant.processResources.manifestFile
            resourcePackageName "com.example.somghosh.earthmiles"
        }
    }

    dependencies {
        compile fileTree(dir: 'libs', include: ['*.jar'])
        apt "org.androidannotations:androidannotations:3.2"
        compile 'org.androidannotations:androidannotations-api:3.2'
        //Core
        compile 'com.github.gabrielemariotti.cards:cardslib-core:2.0.1'
        //Optional for built-in cards
        compile 'com.github.gabrielemariotti.cards:cardslib-cards:2.0.1'
        //Optional for RecyclerView
        compile 'com.github.gabrielemariotti.cards:cardslib-recyclerview:2.0.1'
        //Optional for staggered grid view
        compile 'com.github.gabrielemariotti.cards:cardslib-extra-staggeredgrid:2.0.1'
        //Optional for drag and drop
        compile 'com.github.gabrielemariotti.cards:cardslib-extra-dragdrop:2.0.1'
        compile 'in.srain.cube:ultra-ptr:1.0.5'
        compile 'it.neokree:MaterialTabs:0.10'
        compile 'com.android.support:support-annotations:21.0.3'
        compile 'com.android.support:support-v4:21.0.3'
        compile 'com.squareup.retrofit:retrofit:1.8.0'
        compile 'com.google.code.gson:gson:2.3.1'
        //    compile 'com.google.code.gson:gson:1.7.2'
        //    compile 'org.parceler:parceler:0.2.15'
        // new for fix
        compile 'org.parceler:parceler-api:0.2.16-SNAPSHOT'
        provided 'org.parceler:parceler:0.2.16-SNAPSHOT'
        compile 'com.squareup.okhttp:okhttp:2.1.0'
        compile 'com.squareup.okhttp:okhttp-urlconnection:2.1.0'
        compile 'com.squareup.picasso:picasso:2.4.0'
        compile 'com.github.siyamed:android-shape-imageview:0.9.2'
        compile 'com.squareup.picasso:picasso:2.4.0'
        provided 'org.projectlombok:lombok:1.14.8'
        apt "org.projectlombok:lombok:1.14.8"
        compile 'com.github.talenguyen:prettysharedpreferences:1.0.2'
        compile('de.keyboardsurfer.android.widget:crouton:1.8.5@aar') {
            // exclusion is not neccessary, but generally a good idea.
            exclude group: 'com.google.android', module: 'support-v4'
        }
        compile 'com.afollestad:material-dialogs:0.4.7'
    //    compile 'commons-beanutils:commons-beanutils-core:1.8.3'
    }

On a Android Studio -->Rebuild, this is the message:

    Executing tasks: [clean, :app:compileDebugSources]

    Configuration on demand is an incubating feature.
    WARNING [Project: :app] variant.getProcessResources() is deprecated. Call it on one of variant.getOutputs() instead.
    WARNING [Project: :app] variant.getProcessResources() is deprecated. Call it on one of variant.getOutputs() instead.
    :app:clean
    :app:preBuild
    :app:preDebugBuild
    :app:checkDebugManifest
    :app:preReleaseBuild
    :app:prepareComAfollestadMaterialDialogs047Library
    :app:prepareComAndroidSupportAppcompatV72103Library
    :app:prepareComAndroidSupportCardviewV72100Library
    :app:prepareComAndroidSupportRecyclerviewV72100Library
    :app:prepareComAndroidSupportSupportV42103Library
    :app:prepareComEtsyAndroidGridLibrary105Library
    :app:prepareComGithubGabrielemariottiCardsCardslibCards201Library
    :app:prepareComGithubGabrielemariottiCardsCardslibCore201Library
    :app:prepareComGithubGabrielemariottiCardsCardslibExtraDragdrop201Library
    :app:prepareComGithubGabrielemariottiCardsCardslibExtraStaggeredgrid201Library
    :app:prepareComGithubGabrielemariottiCardsCardslibRecyclerview201Library
    :app:prepareComGithubMarkushiAndroidUi12Library
    :app:prepareComGithubSiyamedAndroidShapeImageview092Library
    :app:prepareDeKeyboardsurferAndroidWidgetCrouton185Library
    :app:prepareInSrainCubeUltraPtr105Library
    :app:prepareItNeokreeMaterialTabs010Library
    :app:prepareDebugDependencies
    :app:compileDebugAidl
    :app:compileDebugRenderscript
    :app:generateDebugBuildConfig
    :app:generateDebugAssets UP-TO-DATE
    :app:mergeDebugAssets
    :app:generateDebugResValues UP-TO-DATE
    :app:generateDebugResources
    :app:mergeDebugResources
    /Users/somghosh/TestAndroid/app/build/intermediates/exploded-aar/com.android.support/appcompat-v7/21.0.3/res/drawable-xhdpi/abc_ab_share_pack_holo_dark.9.png: libpng warning: iCCP: Not recognizing known sRGB profile that has been edited
    /Users/somghosh/TestAndroid/app/build/intermediates/exploded-aar/com.android.support/appcompat-v7/21.0.3/res/drawable-hdpi/abc_spinner_mtrl_am_alpha.9.png: libpng warning: iCCP: Not recognizing known sRGB profile that has been edited
    /Users/somghosh/TestAndroid/app/build/intermediates/exploded-aar/com.android.support/appcompat-v7/21.0.3/res/drawable-mdpi/abc_ab_share_pack_holo_dark.9.png: libpng warning: iCCP: Not recognizing known sRGB profile that has been edited
    /Users/somghosh/TestAndroid/app/build/intermediates/exploded-aar/in.srain.cube/ultra-ptr/1.0.5/res/drawable-xhdpi-v4/ptr_rotate_arrow.png: libpng warning: iCCP: Not recognizing known sRGB profile that has been edited
    /Users/somghosh/TestAndroid/app/build/intermediates/exploded-aar/com.android.support/appcompat-v7/21.0.3/res/drawable-hdpi/abc_ab_share_pack_holo_light.9.png: libpng warning: iCCP: Not recognizing known sRGB profile that has been edited
    /Users/somghosh/TestAndroid/app/build/intermediates/exploded-aar/com.android.support/appcompat-v7/21.0.3/res/drawable-hdpi/abc_ab_share_pack_holo_dark.9.png: libpng warning: iCCP: Not recognizing known sRGB profile that has been edited
    /Users/somghosh/TestAndroid/app/build/intermediates/exploded-aar/com.android.support/appcompat-v7/21.0.3/res/drawable-xhdpi/abc_ab_share_pack_holo_light.9.png: libpng warning: iCCP: Not recognizing known sRGB profile that has been edited

    :app:processDebugManifest
    :app:processDebugResources
    :app:generateDebugSources
    :app:compileDebugJava
    Note: Processing class AccountData
    Note: Processing class FeedDBTableRow
    Note: Processing class GlobalPrefsDBTableRow
    Note: Processing class RewardDBTableRow
    Note: Processing class RewardPartnerDBTableRow
    Note: Resolve log file to /Users/somghosh/TestAndroid/app/build/source/apt/androidannotations.log
    Note: Initialize AndroidAnnotations 3.2 with options {androidManifestFile=/Users/somghosh/TestAndroid/app/build/intermediates/manifests/full/debug/AndroidManifest.xml, resourcePackageName=com.example.somghosh.earthmiles}
    Note: Start processing for 6 annotations on 56 elements
    Note: AndroidManifest.xml file found with specified path: /Users/somghosh/TestAndroid/app/build/intermediates/manifests/full/debug/AndroidManifest.xml
    warning: A class activity declared in the AndroidManifest.xml cannot be found in the compile path: [android.support.v7.widget.TestActivity]
    Note: AndroidManifest.xml found: AndroidManifest [applicationPackage=com.example.somghosh.earthmiles, componentQualifiedNames=[com.example.somghosh.earthmiles.views.MainActivity_], permissionQualifiedNames=[android.permission.INTERNET, android.permission.USE_CREDENTIALS, android.permission.GET_ACCOUNTS, android.permission.READ_PROFILE, android.permission.READ_CONTACTS], applicationClassName=com.example.somghosh.earthmiles.globalstate.App, libraryProject=false, debugabble=false, minSdkVersion=15, maxSdkVersion=-1, targetSdkVersion=21]
    Note: Found project R class: com.example.somghosh.earthmiles.R
    Note: Found Android class: android.R
    Note: Validating elements
    Note: Validating with EActivityHandler: [com.example.somghosh.earthmiles.views.MainActivity]
    Note: Validating with EFragmentHandler: [com.example.somghosh.earthmiles.views.mainthreetabs.mefragment.MeFragment, com.example.somghosh.earthmiles.views.mainthreetabs.newsfragment.NewsFragment, com.example.somghosh.earthmiles.views.mainthreetabs.rewardsfragment.RewardsFragment]
    Note: Validating with EViewGroupHandler: [com.example.somghosh.earthmiles.views.mainthreetabs.newsfragment.NewsRowView]
    Note: Validating with ViewByIdHandler: [viewPager, materialTabHost, userImage, userName, feedText, createdTime, likeCount, commentCount, linearLayoutLike, imageViewLike, imageViewComment]
    Note: Validating with OptionsMenuHandler: [com.example.somghosh.earthmiles.views.MainActivity]
    Note: Validating with AfterViewsHandler: [afterViews()]
    Note: Processing root elements
    Note: Processing root elements EActivityHandler: [com.example.somghosh.earthmiles.views.MainActivity]
    Note: Processing root elements EFragmentHandler: [com.example.somghosh.earthmiles.views.mainthreetabs.mefragment.MeFragment, com.example.somghosh.earthmiles.views.mainthreetabs.newsfragment.NewsFragment, com.example.somghosh.earthmiles.views.mainthreetabs.rewardsfragment.RewardsFragment]
    Note: Processing root elements EViewGroupHandler: [com.example.somghosh.earthmiles.views.mainthreetabs.newsfragment.NewsRowView]
    Note: Processing enclosed elements
    Note: Number of files generated by AndroidAnnotations: 5
    Note: Writting following API classes in project: []
    Note: Generating class: com.example.somghosh.earthmiles.views.MainActivity_
    Note: Generating class: com.example.somghosh.earthmiles.views.mainthreetabs.rewardsfragment.RewardsFragment_
    Note: Generating class: com.example.somghosh.earthmiles.views.mainthreetabs.mefragment.MeFragment_
    Note: Generating class: com.example.somghosh.earthmiles.views.mainthreetabs.newsfragment.NewsFragment_
    Note: Generating class: com.example.somghosh.earthmiles.views.mainthreetabs.newsfragment.NewsRowView_
    Note: Time measurements: [Whole Processing = 288 ms], [Find R Classes = 57 ms], [Validate Annotations = 52 ms], [Process Annotations = 49 ms], [Generate Sources = 48 ms], [Extract Manifest = 27 ms], [Extract Annotations = 25 ms],
    Note: Finish processing
    Note: Start processing for 0 annotations on 11 elements
    Note: Time measurements: [Whole Processing = 0 ms],
    Note: Finish processing
    Note: Start processing for 0 annotations on 0 elements
    Note: Time measurements: [Whole Processing = 1 ms],
    Note: Finish processing
    Note: Start processing for 0 annotations on 0 elements
    Note: Time measurements: [Whole Processing = 1 ms],
    Note: Finish processing
    Note: Some input files use or override a deprecated API.
    Note: Recompile with -Xlint:deprecation for details.
    Note: /Users/somghosh/TestAndroid/app/src/main/java/com/example/somghosh/earthmiles/utils/ToolBox.java uses unchecked or unsafe operations.
    Note: Recompile with -Xlint:unchecked for details.
    1 warning

    :app:compileDebugNdk
    :app:compileDebugSources

    BUILD SUCCESSFUL

    Total time: 25.745 secs

My MainActivity is defined as:

    @EActivity(R.layout.activity_main)
    @OptionsMenu(R.menu.menu_main_actionbar)
    public class MainActivity extends ActionBarActivity implements MaterialTabListener {
    
    
        MainThreeTabAdapter mPagerAdapter;
    
        @ViewById
        MainThreeTabViewPager viewPager;       //automatically gets R.id.viewPager from xml as long as xml name = variable name
    
        @ViewById
        MaterialTabHost materialTabHost;
    
        ActionBar actionBar;
    
    
        private Resources res;
        private String[] tabStrings = { "News", "Rewards", "Me" };
    
        @AfterViews
        void afterViews() {
            actionBar = getSupportActionBar();
            actionBar.setDisplayOptions(ActionBar.DISPLAY_SHOW_HOME | ActionBar.DISPLAY_SHOW_TITLE);
            actionBar.setDisplayShowHomeEnabled(true);
            actionBar.setDisplayUseLogoEnabled(true);
            actionBar.setLogo(R.drawable.green_drawable);
    
            res = this.getResources();      // assign res to be used later for finding icons
            mPagerAdapter = new MainThreeTabAdapter(getSupportFragmentManager());
            viewPager.setAdapter(mPagerAdapter);
    
            viewPager.setOnPageChangeListener(new MainThreeTabViewPager.SimpleOnPageChangeListener() {
                @Override
                public void onPageSelected(int position) {
                    // when user does a swipe the selected tab change
                    materialTabHost.setSelectedNavigationItem(position);
                }
            });
    
            // add the tab titles
            for (String tab : tabStrings) {
                materialTabHost.addTab(
                        materialTabHost.newTab()
                                .setText(tab)
                                .setTabListener(this)
                );
            }
            
            
    
            
        }
    
        @Override
        public void onTabSelected(MaterialTab materialTab) {
            viewPager.setCurrentItem(materialTab.getPosition());
        }
    
        @Override
        public void onTabReselected(MaterialTab materialTab) { }
    
        @Override
        public void onTabUnselected(MaterialTab materialTab) { }
    
    
    
    }

**Comment by WonderCsabo** [comment]
No answer since 10 days, closing the issue. Please feel free to reopen if you have something to share.

**Comment by HermannK** [comment]
I ran into the same issue (see #1320). 
I added AA to an existing project which worked fine for all but the main activity. Same as described above. It turned out that I did not re-import my project into AndroidStudio after the build.gradle files where changed properly. 
See https://github.com/excilys/androidannotations/wiki/Gradle-IntelliJ 
Apparently is is important to actually open the very build.gradle file and go from there. Then "Make" the project (CTRL-F9).
